### PR TITLE
MULE-16500: WeakReferences are retained in a non weakreference set in CursorManager

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

**Comment by fsgonz** [comment]
I'll leave change the commit on squash so that MULE-16500 is referenced.

File: core/src/main/java/org/mule/runtime/core/internal/streaming/CursorManager.java [hunk 1]
```diff
@@ -21,7 +22,9 @@
 import com.github.benmanes.caffeine.cache.LoadingCache;
 
 import java.lang.ref.WeakReference;
+import java.util.Collections;
```
Comment (sonar-mulesoft) [inline core/src/main/java/org/mule/runtime/core/internal/streaming/CursorManager.java]: Code Smell Code Smell: Remove this unused import 'java.util.Collections'. ([squid:UselessImportCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7688&issues=AWj9Gu0AotIXnmsUljPh&open=AWj9Gu0AotIXnmsUljPh)

[ISSUE_KEY]: AWj9Gu0AotIXnmsUljPh

File: core/src/main/java/org/mule/runtime/core/internal/streaming/CursorManager.java [hunk 2]
```diff
@@ -21,7 +22,9 @@
...
 import java.util.Set;
+import java.util.WeakHashMap;
 import java.util.concurrent.ConcurrentHashMap;
```
Comment (sonar-mulesoft) [inline core/src/main/java/org/mule/runtime/core/internal/streaming/CursorManager.java]: Code Smell Code Smell: Remove this unused import 'java.util.concurrent.ConcurrentHashMap'. ([squid:UselessImportCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7688&issues=AWj9Gu0AotIXnmsUljPg&open=AWj9Gu0AotIXnmsUljPg)

[ISSUE_KEY]: AWj9Gu0AotIXnmsUljPg

**Comment by fsgonz** [comment]
This PR is incorrect. It is not needed. The janitor should be the one to take care of releasing the resources, so it is correct to have a strong reference to a weak reference.

**Pull request closed without merging**